lib/facility.rb

```diff
@@ -1,14 +1,73 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles
+  attr_accessor :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def set_plate_type(car)
+    if car.antique? == true
+      :antique
+    elsif car.electric_vehicle? == true
+      :ev
+    else 
+      :regular
+    end
+  end
+
+  #helper method
+  def check_for_service(service)
+    @services.include?(service)
+  end
+
+  def register_vehicle(car)
+    if check_for_service('Vehicle Registration') == true
```

Comment:
This is a really cool use of a helper method. A couple of thoughts:
1. You could omit the `== true` since `check_for_service('Vehicle Registration')` will already return a boolean, which would work on its own for the conditional logic. 
2. Per convention, you could define the helper method with a `?` at the end of the name to indicate it will return a boolean value.
```
def has_service?(service)
  ...
end
```
3. By using a guard clause, you could avoid the need to nest conditional statements in this method:
```
def register_vehicle(car)
  return false unless has_service?('Vehicle Registration')
  @registered_vehicles << car
  ...
```